File: Hash Table/3396.Minimum Number of Operations to Make Elements in Array Distinct/Solution.py
class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        n = len(nums)
        s = set()
        for i in range(n - 1, -1, -1):
            if nums[i] in s:
                return i // 3 + 1
            s.add(nums[i])
        return 0

File: Hash Table/3396.Minimum Number of Operations to Make Elements in Array Distinct/Solution.java
class Solution {
    public int minimumOperations(int[] nums) {
        int n = nums.length;
        Set<Integer> s = new HashSet<>();
        for (int i = n - 1; i > -1; i--) {
            if (s.contains(nums[i])) {
                return i / 3 + 1;
            }
            s.add(nums[i]);
        }
        return 0;
    }
}